tests/api/v1/auditEvents/get.js

```diff
@@ -182,4 +181,274 @@ describe('tests/api/v1/auditEvents/get.js >', () => {
+        expect(res.body[2].resourceName).to.equal('Sample');
+        return done();
+      });
+    });
+  });
+
+  describe('date filtering with relativeDateTime', () => {
+    const date = new Date();
+
+    // set date to current date + 10 minutes
+    date.setMinutes(date.getMinutes() + 10);
+    const aeWith10MinOffset = u.createAuditEventObject('CollectorResource',
+      'Heartbeat', date, 'true');
+
+    beforeEach((done) => {
+      AuditEvent.create(aeWith10MinOffset)
+      .then(() => done())
+      .catch(done);
+    });
+
+    afterEach(u.forceDelete);
+
+    it('Not Ok, relativeDateTime must end with a time unit', (done) => {
+      api.get(`${path}?relativeDateTime=-15`)
+      .set('Authorization', token)
+      .expect(constants.httpStatus.BAD_REQUEST)
+      .end((err, res) => {
+        if (err) {
+          return done(err);
+        }
+
+        expect(res.body.errors[0].description).to
+        .include('RelativeDateTime should start with a negation (-) ' +
+          'and end with a valid time unit. The valid time units are d,h,m,s');
+        return done();
+      });
+    });
+
+    it('Not Ok, relativeDateTime should have the right time unit', (done) => {
+      api.get(`${path}?relativeDateTime=-15a`)
+      .set('Authorization', token)
+      .expect(constants.httpStatus.BAD_REQUEST)
+      .end((err, res) => {
+        if (err) {
+          return done(err);
+        }
+
+        expect(res.body.errors[0].description).to
+        .include('RelativeDateTime should start with a negation (-) ' +
+          'and end with a valid time unit. The valid time units are d,h,m,s');
+        return done();
+      });
+    });
+
+    it('Not Ok, should begin with - ', (done) => {
+      api.get(`${path}?relativeDateTime=15m`)
+      .set('Authorization', token)
+      .expect(constants.httpStatus.BAD_REQUEST)
+      .end((err, res) => {
+        if (err) {
+          return done(err);
+        }
+
+        expect(res.body.errors[0].description).to
+        .include('RelativeDateTime should start with a negation (-) ' +
+          'and end with a valid time unit. The valid time units are d,h,m,s');
+        return done();
+      });
+    });
+
+    it('Not Ok, invalid offset in relativeDateTime', (done) => {
+      api.get(`${path}?relativeDateTime=-invalid`)
+      .set('Authorization', token)
+      .expect(constants.httpStatus.BAD_REQUEST)
+      .end((err, res) => {
+        if (err) {
+          return done(err);
+        }
+
+        expect(res.body.errors[0].message).to.include('Invalid date');
+        return done();
+      });
+    });
+
+    it('Ok, relativeDateTime filter with seconds', (done) => {
+      const d = new Date();
+
+      // set date to current date + 3 seconds
+      d.setSeconds(d.getSeconds() + 3);
+      const ae = u.createAuditEventObject('Collector', 'GeneratorTemplate',
+        d, 'true');
+      AuditEvent.create(ae)
+      .then(() => {
+        api.get(`${path}?relativeDateTime=-s`)
```

Comment:
relativeDateTime=-s Is this valid? If not, do we need validation that there should be a number?

Reply:
relativeDateTime=-s or -h or -d or -m are converted to relativeDateTime=-0s .. Let me know if we want to validate it or not.